sunpy/map/mapbase.py

```diff
@@ -993,11 +993,11 @@ def rotate(self, angle=None, rmatrix=None, order=4, scale=1.0,
         new_map.meta['crpix2'] = new_reference_pixel[1] + 1 # FITS pixel origin is 1
 
         # Unpad the array if necessary
-        unpad_x = -np.min((diff[1], 0))
+        unpad_x = -int(np.min((diff[1], 0)))
         if unpad_x > 0:
             new_map.data = new_map.data[:, unpad_x:-unpad_x]
             new_map.meta['crpix1'] -= unpad_x
-        unpad_y = -np.min((diff[0], 0))
+        unpad_y = -int(np.min((diff[0], 0)))
```

Comment:
would make sense to make `int` the `diff` which is using a `np.ceil` so it should be ok to set it to integer (l950)